presentation/src/main/java/com/example/pinterbest/presentation/CreatorsFragment.kt

```diff
@@ -33,5 +52,54 @@ class CreatorsFragment : Fragment() {
                 .findFragmentById(R.id.NavHostFragment) as NavHostFragment
             navHostFragment.navController.navigateUp()
         }
+
+        creatorsAdapter = CreatorsAdapter()
+        binding.rvCreators.apply {
+            adapter = creatorsAdapter
+            layoutManager = LinearLayoutManager(
+                requireContext(),
+                LinearLayoutManager.VERTICAL,
+                false
+            )
+        }
+
+        initObservers()
+    }
+
+    private fun initObservers() {
+        viewModel.creators.observe(
+            viewLifecycleOwner
+        ) {
+            it?.let {
+                showCreators(it)
+            }
+        }
+
+        viewModel.error.observe(
+            viewLifecycleOwner
+        ) {
+            showError(it)
+        }
+
+        viewModel.state.observe(
+            viewLifecycleOwner
+        ) { loading ->
+            when (loading) {
+                true -> binding.progressBar.visibility = View.VISIBLE
+                false -> binding.progressBar.visibility = View.GONE
+            }
```

Comment:
again, I would suggest to replace with `binding.progressBar.visibility = if (loading) View.VISIBLE else View.GONE`

Reply:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304